### PR TITLE
IPv4/single: applied MISRA to FreeRTOS_UDP_IP.c

### PR DESCRIPTION
**Pull request by htibosch**
From time to time we check if the code is still complying to the MISRA rules.

This PR presents some related change for FreeRTOS_UDP_IP.c.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: include/FreeRTOS_IP_Private.h [hunk 1]
```diff
@@ -48,16 +48,18 @@
 /*-----------------------------------------------------------*/
 /* Utility macros for marking casts as recognized during     */
 /* static analysis.                                          */
+/* Note _HT_ Changed 'vCastConstPointerTo' to the shorter    */
+/* vCastConstPtrTo to limit the length of the function name. */
```
Comment (AniruddhaKanhere) [inline include/FreeRTOS_IP_Private.h:52]: Why is name length required to be limited?

File: include/FreeRTOS_IP_Private.h [hunk 2]
```diff
@@ -48,16 +48,18 @@
...
 /*-----------------------------------------------------------*/
     #define ipCAST_PTR_TO_TYPE_PTR( TYPE, pointer )                ( vCastPointerTo_ ## TYPE( ( void * ) ( pointer ) ) )
-    #define ipCAST_CONST_PTR_TO_CONST_TYPE_PTR( TYPE, pointer )    ( vCastConstPointerTo_ ## TYPE( ( const void * ) ( pointer ) ) )
+    #define ipCAST_CONST_PTR_TO_CONST_TYPE_PTR( TYPE, pointer )    ( vCastConstPtrTo_ ## TYPE( ( const void * ) ( pointer ) ) )
```
Comment (AniruddhaKanhere) [inline include/FreeRTOS_IP_Private.h:55]: Same question.

File: include/FreeRTOS_IP.h [hunk 3]
```diff
@@ -350,6 +361,11 @@
  * be defined in a user module. */
     extern BaseType_t xApplicationGetRandomNumber( uint32_t * pulNumber );
```
Comment (AniruddhaKanhere) [inline include/FreeRTOS_IP.h:362]: I don't think extern is required
```suggestion
    BaseType_t xApplicationGetRandomNumber( uint32_t * pulNumber );
```

File: include/FreeRTOS_IP_Private.h (hunk 1 above)
Comment (htibosch) [reply include/FreeRTOS_IP_Private.h:52]: MISRA likes to make sure that all identifiers are still unique if only the first 32 characters are used. This is true for both macros, variables and function names.

One example is `vCastConstPointerTo_ICMPRouterAdvertisement_IPv6_t()` with 50 characters.

**Review by AniruddhaKanhere — COMMENTED** [review]
Thank you @htibosch for this PR.
If these changes are also merged in the single all encompassing PR, then I think we can close this.

**Comment by htibosch** [comment]
The changes proposed in this PR also appear in PR #357, and they will be merged to main from there.
Closing this PR now.

**Pull request closed without merging**